packages/api/amplify_api/example/integration_test/graphql/iam_test.dart

```diff
@@ -443,6 +443,41 @@ void main({bool useExistingTestUser = false}) {
           expect(postFromEvent?.title, equals(title));
         });
 
+        testWidgets('should support where clause when using Model helpers',
```

Comment:
I think to really test this you should create two blogs and two posts:

```dart
final blog1 = Blog(name: 'blog 1');
final blog2 = Blog(name: 'blog 2');
final post1 = Post(title: 'post 1', blog: blog1);
final post2 = Post(title: 'post 2', blog: blog2);

final sub = subscribe(Post.classType, where: Post.BLOG.eq(blog2.id));

addBlog(blog1);
addBlog(blog2);
addPost(post1);
addPost(post2);
```

`where` should filter out the first post and only return the second.

Currently, this could just be returning the first post.